### PR TITLE
Battle tower data

### PR DESCRIPTION
**Pull request by garakmon**

### CODE REVIEW

**Comment by DizzyEggg** [comment]
@Diegoisawesome

File: include/constants/trainers.h [hunk 1]
```diff
@@ -200,6 +200,8 @@
 #define FACILITY_CLASS_RS_BRENDAN           0x50
 #define FACILITY_CLASS_RS_MAY               0x51
 
+#define FACILITY_CLASSES_COUNT 0x52
```
Comment (Diegoisawesome) [inline include/constants/trainers.h]: Align properly

File: src/battle_tower.c [hunk 2]
```diff
@@ -102,6 +76,982 @@ static void SetEReaderTrainerChecksum(struct BattleTowerEReaderTrainer *ereaderT
...
+    BattleFrontier_BattleTowerMultiBattleRoom_Text_248D04
+};
+
+struct
+{
+    u32 facilityClass;
+    const u8 *const *strings;
+} const gUnknown_085DD500[] =
+{
+    {0x00000047, gUnknown_085DD118},
```
Comment (Diegoisawesome) [inline src/battle_tower.c]: Use constants

File: src/battle_tower.c [hunk 3]
```diff
@@ -147,7 +147,7 @@ const u16 gBattleFrontierHeldItems[] =
 #include "data/battle_frontier/battle_frontier_trainers.h"
 #include "data/battle_frontier/battle_frontier_mons.h"
 
-const u8 gTowerMaleFacilityClasses[] = 
+const u8 gTowerMaleFacilityClasses[30] =
```
Comment (Meowsy) [inline src/battle_tower.c]: Is there any point to specifying the length of the arrays `gTowerMaleFacilityClasses`, `gTowerFemaleFacilityClasses`, `gTowerMaleTrainerGfxIds`, and `gTowerFemaleTrainerGfxIds`? The compiler should be able to figure that out automatically.

Comment (huderlem) [reply src/battle_tower.c]: yeah, they should be removed.  They are only useful when declaring them as extern.

Comment (DizzyEggg) [reply src/battle_tower.c]: They have to be there, otherwise you won't be able to extern it while specyfing the array length.

**Comment by garakmon** [comment]
fixed all the things

File: src/battle_tower.c (hunk 3 above)
Comment (Diegoisawesome) [reply src/battle_tower.c]: What do you mean? If you need to extern it from another file then you can do so without specifying the length.

Comment (Diegoisawesome) [reply src/battle_tower.c]: Apparently removing these causes Travis to have an internal compiler error?

Comment (garakmon) [reply src/battle_tower.c]: I guess it has something to do with apprentice.c using ARRAY_COUNT on this array but it not being defined in that file so the length needs to be explicit.

**Pull request merged**